pyproject.toml

```diff
@@ -6,6 +6,35 @@ build-backend = "setuptools.build_meta"
 line-length = 100
 target-version = ['py38', 'py39', 'py310', 'py311']
 
+[tool.ruff]
+line-length = 100
+src = ["rustworkx", "setup.py", "retworkx", "tests"]
+select = [
+    "E",   # pycodestyle
+    "F",   # pyflakes
+    "UP",  # pyupgrade
+    "PYI", # flake8-pyi
+    "Q",   # flake8-quotes
+]
+target-version = "py38"
+exclude = [
+    ".venv",
+    ".git",
+    ".tox",
+    ".eggs",
+    "dist",
+    "doc",
+    "build",
+]
+
+[tool.ruff.per-file-ignores]
+"rustworkx/__init__.py" = ["F405", "F403"]
+"rustworkx/__init__.pyi" = ["F403", "F405", "PYI001"]
+"rustworkx/digraph.pyi" = ["F403", "F405", "PYI001"]
+"rustworkx/graph.pyi" = ["F403", "F405", "PYI001"]
+"rustworkx/iterators.pyi" = ["F403", "F405", "PYI001"]
```

Comment:
I should fix `PYI001` in those files, it would avoid us having to define `__all__` in them